src/jit/lsraxarch.cpp

```diff
@@ -2949,34 +2693,69 @@ void Lowering::TreeNodeInfoInitGCWriteBarrier(GenTree* tree)
 // Arguments:
 //    indirTree    -   GT_IND or GT_STOREIND gentree node
 //
-void Lowering::TreeNodeInfoInitIndir(GenTreePtr indirTree)
+void Lowering::TreeNodeInfoInitIndir(GenTreeIndir* indirTree)
 {
-    assert(indirTree->isIndir());
     // If this is the rhs of a block copy (i.e. non-enregisterable struct),
     // it has no register requirements.
     if (indirTree->TypeGet() == TYP_STRUCT)
     {
         return;
     }
 
-    GenTreePtr    addr = indirTree->gtGetOp1();
+    ContainCheckIndir(indirTree);
+
+    GenTree*      addr = indirTree->gtGetOp1();
     TreeNodeInfo* info = &(indirTree->gtLsraInfo);
 
     GenTreePtr base  = nullptr;
     GenTreePtr index = nullptr;
     unsigned   mul, cns;
     bool       rev;
 
+    bool isStore   = (indirTree->gtOper == GT_STOREIND);
```

Comment:
Is `isStore` used anywhere besides the `if` condition two lines down? If not, might as well get rid of it.